model/task_history.go

```diff
@@ -394,17 +390,35 @@ func (self *taskHistoryIterator) GetDistinctTestNames(numCommits int) ([]string,
 			{"$unwind": fmt.Sprintf("$%v", testResultsKey)},
 			{"$group": bson.M{"_id": fmt.Sprintf("$%v.%v", testResultsKey, task.TestResultTestFileKey)}},
 		},
+		opts,
 	)
-
-	var output []bson.M
-
-	if err = pipeline.All(&output); err != nil {
-		return nil, errors.WithStack(err)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting cursor")
+	}
+	if cursor == nil {
+		return nil, errors.New("nil cursor returned")
 	}
 
-	names := make([]string, 0)
-	for _, doc := range output {
-		names = append(names, doc["_id"].(string))
+	names := []string{}
+	for cursor.Next(ctx) {
+		select {
+		case <-ctx.Done():
+			_ = cursor.Close(ctx)
+			return nil, errors.New("context cancelled")
+		default:
+			// The cursor API expects us to _copy_ the current data if we intend to use
```

Comment:
i copied this comment word for word from May's code haha but I can take it out if we don't like that 😂

Reply:
It would seem to me like since we know what the documents will look like we can use the cursor's Decode function to implicitly copy into a `bson.M`

Reply:
you are correct